spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -88,82 +93,247 @@ public void setTaskExecutor(TaskExecutor taskExecutor) {
 	/**
 	 * Each queue runs in its own consumer; set this property to create multiple
 	 * consumers for each queue.
+	 * If the container is already running, the number of consumers per queue will
+	 * be adjusted up or down as necessary.
+	 * If the adjustment fails for any reason, you may have mismatched number of
+	 * consumers on each queue. A subsequent (successful) call will fix the mismatch.
 	 * @param consumersPerQueue the consumers per queue.
 	 */
 	public void setConsumersPerQueue(int consumersPerQueue) {
+		Assert.isTrue(consumersPerQueue >= 1, "'consumersPerQueue' must be 1 or greater");
+		if (isRunning()) {
+			adjustConsumers(consumersPerQueue);
+		}
 		this.consumersPerQueue = consumersPerQueue;
 	}
 
 	/**
-	 * Set to false to requeue failed deliveries.
-	 * @param requeueRejected the requeue rejected.
+	 * Set to true for an exclusive consumer - if true, the
+	 * {@link #setConsumersPerQueue(int) consumers per queue} must be 1.
+	 * @param exclusive true for an exclusive consumer.
+	 */
+	@Override
+	public final void setExclusive(boolean exclusive) {
+		Assert.isTrue(!exclusive || (this.consumersPerQueue == 1),
+				"When the consumer is exclusive, the consumers per queue must be 1");
+		super.setExclusive(exclusive);
+	}
+
+	/**
+	 * Set the interval before retrying when starting the container; default 5000.
+	 * @param recoveryInterval the interval.
 	 */
-	public void setDefaultRequeueRejected(boolean requeueRejected) {
-		this.defaultRequeueRejected = requeueRejected;
+	public void setRecoveryInterval(long recoveryInterval) {
+		this.recoveryInterval = recoveryInterval;
+	}
+
+	@Override
+	public void setQueueNames(String... queueName) {
+		Assert.state(!isRunning(), "Cannot set queue names while running, use add/remove");
+		super.setQueueNames(queueName);
+	}
+
+	@Override
+	public void setQueues(Queue... queues) {
+		Assert.state(!isRunning(), "Cannot set queue names while running, use add/remove");
+		super.setQueues(queues);
+	}
+
+	@Override
+	public void addQueueNames(String... queueNames) {
+		try {
+			addQueues(Arrays.stream(queueNames));
+		}
+		catch (AmqpIOException e) {
+			throw new AmqpIOException("Failed to add " + queueNames, e.getCause());
```

Comment:
Ha! I think this would be even better. 
We definitely need only a string representation and there is no reason in that intermediate `ArrayList`  object.

Good catch, @kartoffelsup !
Thank you!